MariaDB/server
# MDEV-19161: Let galera_new_cluster use "restart" instead of "start"

-- [x] *The Jira issue number for this PR is: MDEV-19161*


## Description

galera_new_cluster so far uses `systemctl start mariadb`, this is a no-op though if the mariadb service is already running, so no new cluster would be started in that case, yet the script call would return success

by changing this to `systemctl restart` instead of `start` it is ensured that a new cluster is created even if the mariadb service was already running beforehand

## How can this PR be tested?

* check that the mariadb service is actually running with `systemctl is-active mariadb`
* run `galera_new_cluster`
* verify that the mariadb process was indeed restarted by e.g. checking its uptime, or the systemd journal

E.g. without the patch:

```
# systemctl is-active mariadb
active

# galera_new_cluster 

# mysql -e "show status like 'Uptime'"
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Uptime        | 435   |
+---------------+-------+
```

With the patch applied:

```# systemctl is-active mariadb
active

# galera_new_cluster 

# mysql -e "show status like 'Uptime'"
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Uptime        | 3     |
+---------------+-------+
```

## Basing the PR against the correct MariaDB version

Right now it is the former, but it may actually make sense to make it the later, fixing it in all supported releases

## Backward compatibility

I don't think that anyone would be relying on the current behavior and so would be negatively affected by this being changed
